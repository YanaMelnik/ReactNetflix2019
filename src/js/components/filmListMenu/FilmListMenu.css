.film-list-menu {
    width: 100%;
    background: #f1eeee;
    min-height: 40px;
    display: flex;
}

.film-list-menu__wrapper {
    display: flex;
    justify-content: space-between;
    font-family: sans-serif;
    font-weight: 500;
    font-size: 14px;
}

.film-list-menu__left {
    margin: 20px 0 20px;
}