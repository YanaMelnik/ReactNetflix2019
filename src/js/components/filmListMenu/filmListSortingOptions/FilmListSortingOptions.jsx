import React from 'react';

const FilmListSortingOptions = () => <form>
        <span>Sort by</span>
        <span>
            <input name='sort__by' type='radio' value='date'/>
            release date
        </span>
        <span>
            <input name='sort__by' type='radio' value='rating' checked/>
            rating
        </span>
    </form>;
export default FilmListSortingOptions;
