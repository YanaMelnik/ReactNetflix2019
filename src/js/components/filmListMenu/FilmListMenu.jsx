import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './FilmListMenu.css';
import FilmListSortingOptions from './filmListSortingOptions/FilmListSortingOptions';

class FilmListMenu extends Component {
  render() {
    let contentElement;
    if (this.props.genreName) {
      contentElement = `Films by ${this.props.genreName} genre`;
    } else if (this.props.resultLength) {
      contentElement = `${this.props.resultLength} movies found`;
    }

    return (
      <div className='film-list-menu'>
        <div className='wrapper film-list-menu__wrapper'>
          <span className='film-list-menu__left'>
            { contentElement }
          </span>
          {this.props.filmSorting
            ? (
              <span className='film-list-menu__right'>
                <FilmListSortingOptions/>
              </span>
            )
            : null
          }
        </div>
      </div>
    );
  }
}

FilmListMenu.propTypes = {
  filmSorting: PropTypes.bool,
  genreName: PropTypes.string,
  resultLength: PropTypes.string,
};

FilmListMenu.defaultProps = {
  filmSorting: false,
};

export default FilmListMenu;
