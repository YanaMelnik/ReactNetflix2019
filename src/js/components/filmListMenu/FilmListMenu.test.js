import React from 'react';
import { shallow, mount } from 'enzyme';
import FilmListMenu from './FilmListMenu';
import FilmListSortingOptions from './filmListSortingOptions/FilmListSortingOptions';

describe('FilmListMenu', () => {
    it('should render component with sorting view and count of film', () => {
        const component = shallow(<FilmListMenu resultLength={'7'} filmSorting={true}/>);

        expect(component).toMatchSnapshot();
    });

    it('should render component with film genre name', () => {
        const component = shallow(<FilmListMenu genreName='drama'/>);

        expect(component).toMatchSnapshot();
    });

    it('should set correct title with film genre name', () => {
        const genreProps = 'drama';
        const genreName = `Films by ${genreProps} genre`;

        const component = mount(<FilmListMenu genreName={genreProps}/>);

        expect(component.find('span.film-list-menu__left').text()).toEqual(genreName);
    });

    it('should set correct title with details of count of film', () => {
        const resultLengthProps = '7';
        const resultLengthText = `${resultLengthProps} movies found`;

        const component = mount(<FilmListMenu resultLength={resultLengthProps}/>);

        expect(component.find('span.film-list-menu__left').text()).toEqual(resultLengthText);
    });

    it('should show sorting option if props is available', () => {
        const filmSorting = true;

        const component = mount(<FilmListMenu filmSorting={filmSorting}/>);

        expect(component.find(FilmListSortingOptions)).toHaveLength(1);
    });
});
