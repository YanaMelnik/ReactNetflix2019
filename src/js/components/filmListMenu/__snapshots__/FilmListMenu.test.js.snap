// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmListMenu should render component with film genre name 1`] = `
<div
  className="film-list-menu"
>
  <div
    className="wrapper film-list-menu__wrapper"
  >
    <span
      className="film-list-menu__left"
    >
      Films by drama genre
    </span>
  </div>
</div>
`;

exports[`FilmListMenu should render component with sorting view and count of film 1`] = `
<div
  className="film-list-menu"
>
  <div
    className="wrapper film-list-menu__wrapper"
  >
    <span
      className="film-list-menu__left"
    >
      7 movies found
    </span>
    <span
      className="film-list-menu__right"
    >
      <Connect(FilmListSortingOptions) />
    </span>
  </div>
</div>
`;
